src/libsystemd/sd-journal/journal-file.c

```diff
@@ -4159,7 +4174,7 @@ int journal_file_copy_entry(JournalFile *from, JournalFile *to, Object *o, uint6
                         return r;
         }
 
-        r = journal_file_append_entry_internal(to, &ts, boot_id, xor_hash, items, n, NULL, NULL, NULL);
+        r = journal_file_append_entry_internal(to, &ts, boot_id, &from->header->machine_id, xor_hash, items, n, NULL, NULL, NULL);
```

Comment:
Does this actually do anything? If we open a journal file, it gets assigned the machine ID from the current machine, and append_entry_internal() will only overwrite it if it is not NULL. So even if the machine ID form the file we're copying is different, it won't actually end up in the journal file.

Reply:
Yes, it does something. See above: the machine ID might not be initialized when the journal file is initially created, and we handle this gracefully after all: as in the code block you commented on earlier. If the machine ID is not set yet we leave it unset, i.e. all zeroes. This code here than tries to fill in the machine ID from the copy source if not set.

This actually matters in the test cases: if run under debian/LXC/ubuntu stuff the machine ID is unset, and this here tries to make something sensible from this situation.